infra/ballista/build/lambda-handler.py

```diff
@@ -0,0 +1,190 @@
+                process_s.terminate()
+                logging.error(process_s.stdout.read().decode("utf-8"))
+                logging.error(process_s.stderr.read().decode("utf-8"))
+                raise Exception(f"executor failed to start after {timeout} seconds")
+    else:
+        logging.error(process_e.stdout.read().decode("utf-8"))
+        logging.error(process_s.stdout.read().decode("utf-8"))
+        raise Exception(
+            f"executor failed to init stderror: {process_e.stderr.read().decode('utf-8')}"
+        )
+
+    global process_cli
+    logging.info("cli starts")
+    process_cli = popen_spawn.PopenSpawn(
+        [
+            "/opt/ballista/ballista-cli",
+            "--host",
+            "localhost",
+            "--port",
+            "50050",
+            "--format",
+            "csv",
+        ]
+    )
+    process_cli.expect(b"\n")
+    logging.debug(process_cli.before)
+
+
+def handler(event, context):
+    """An AWS Lambda handler that runs the provided command with bash and returns the standard output"""
+    global IS_COLD_START
+    is_cold_start = IS_COLD_START
+    IS_COLD_START = False
+
+    start = time.time()
+
+    if is_cold_start:
+        init()
+    init_duration = time.time() - start
+
+    # input parameters
+    logging.debug("event: %s", event)
+    src_command = base64.b64decode(event["cmd"]).decode("utf-8")
+
+    logging.info("command: %s", src_command)
+    if "env" in event:
+        logging.info("env: %s", event["env"])
+        for (k, v) in event["env"].items():
+            os.environ[k] = v
+
+    # process_cli = subprocess.run(command, capture_output=True)
+    global process_cli
+    tmp_out = b""
+    tmp_error = b""
+    try:
+        for command in src_command.split(";"):
+            if command != "":
+                try:
+                    process_cli.sendline(command + ";")
+                    logging.debug(command)
+                    look_for = [
+                        b"Query took",
+                        pexpect.EOF,
+                        b'DataFusionError\(Execution\("Table .*already exists"\)\)',
+                    ]
+                    i = process_cli.expect(look_for, timeout=60)
+                    tmp_out = tmp_out + process_cli.before
+                    if i == 0:
+                        tmp_out += look_for[i]
+                    elif i == 2:
+                        tmp_error += look_for[i]
+                        break
+                    else:
+                        tmp_error += "pexpect.exceptions.EOF: End Of File (EOF)"
+                except pexpect.exceptions.TIMEOUT:
+                    logging.debug("this command timeout flushing std_out to output")
+                    tmp_out += process_cli.before
+    finally:
+        process_cli.expect(b"\n")
+        tmp_out = tmp_out + process_cli.before + b"\n"
+
+    cli_stdout = tmp_out.decode("utf-8")
+    if tmp_error != b"":
+        ret_code = 1
+    else:
+        ret_code = 0
```

Comment:
⁉️ with regard to how the monitoring [now works](https://github.com/cloudfuse-io/lambdatization/pull/31), it is better to raise if an error occurred. Otherwise we take the risk of recording the execution in our metrics and the time measured might be off. I also fixed Dremio in that regard in https://github.com/cloudfuse-io/lambdatization/pull/41

Reply:
refactored on a97d804d